test/officerManager.js

```diff
@@ -0,0 +1,336 @@
+/* global describe it before after */
+// During the test the env variable is set to test
+process.env.NODE_ENV = 'test'
+const OfficerManager = require('../api/models/OfficerManager')
+const User = require('../api/models/User')
+// Require the dev-dependencies
+const chai = require('chai')
+const chaiHttp = require('chai-http')
+const server = require('../server')
+let serverInstance = null
+let app = null
+const expect = chai.expect
+// tools for testing
+const tools = require('../util/testing-utils/tools.js')
+
+chai.should()
+chai.use(chaiHttp)
+
+function initializeServer () {
+  serverInstance = new server.Server()
+  serverInstance.openConnection()
+  app = serverInstance.getServerInstance()
+}
+
+function terminateServer (done) {
+  serverInstance.closeConnection(done)
+}
+
+// Our parent block
+describe('OfficerManager', () => {
+  before(done => {
+    initializeServer(app)
+
+    // Before each test we empty the database
+    tools.emptySchema(OfficerManager)
+    tools.emptySchema(User)
+    done()
+  })
+  after(done => {
+    terminateServer(done)
+  })
+
+  let token = ''
+
+  describe('/POST submit', () => {
+    it('Should register a user', done => {
+      const user = {
+        email: 'test@test.com',
+        password: 'pass',
+        firstName: 'first-name',
+        lastName: 'last-name'
+      }
+
+      chai
+        .request(app)
+        .post('/api/user/register')
+        .send(user)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should log a user in and get a token', done => {
+      const user = {
+        email: 'test@test.com',
+        password: 'pass'
+      }
+      chai
+        .request(app)
+        .post('/api/user/login')
+        .send(user)
+        .then(function (res) {
+          expect(res).to.have.status(200)
+          res.body.should.be.a('object')
+          res.body.should.have.property('token')
+          token = res.body.token
+
+          done()
+        })
+        .catch(err => {
+          throw err
+        })
+    })
+
+    it('Should not return statusCode 200 when the required fields are not set', done => {
+      const form = {}
+      chai
+        .request(app)
+        .post('/api/officerManager/submit')
+        .send(form)
+        .then(function (res) {
+          expect(res).to.not.have.status(200)
```

Comment:
change the field to be `BAD_REQUEST`. MongoDB APIs can also return 500 in some situations and it is best to test the defined behavior.

Reply:
changed to 400